dptb/tests/test_trainer_hardening.py

```
@@ -0,0 +1,386 @@
+    assert moved > 0.5, f"warmup step should track fast, moved {moved}"
+
+
+def test_ema_state_dict_roundtrip_and_mismatch():
+    torch.manual_seed(0)
+    m = torch.nn.Linear(3, 3)
+    ema = ExponentialMovingAverage(m.parameters(), decay=0.9)
+    ema.update()
+    sd = ema.state_dict()
+    ema2 = ExponentialMovingAverage(m.parameters(), decay=0.9)
+    ema2.load_state_dict(sd)
+    for a, b in zip(ema.shadow, ema2.shadow):
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)